### DISCUSSION TITLE
Personal Access Tokens in a Gradle project stop working after uploading to a Maven repo

### DISCUSSION THREAD

**Discussion by felix-t-morgenstern** [community/community#38128] [Actions]
The title says the gist of it... I have a number of repositories, and I upload them as Packages, using Personal Access Tokens. Most of my projects are written in Maven, and when I'm using Maven, all of my PATs work just fine. However, when I want to download dependencies and push package updates in my Gradle project, the PAT only works _until I upload a package version with one of my Maven projects_. The PAT for my Maven projects still works totally fine, but once I use _any_ PAT on _any_ Maven project, _none_ of my existing PATs will work with Gradle at that point, so I have to (as far as I know) create a new one. I have no idea what would be causing this; my build.gradle file for the only project I have in Gradle looks like so:

`plugins {
    id('java')
    id("maven-publish")
}

group 'disorg.soliloquy'
version '0.0.0'

repositories {
    mavenCentral()
    maven {
        url = uri("https://maven.pkg.github.com/felix-t-morgenstern/soliloquy.specs")
        credentials {
            username = 'felix-t-morgenstern'
            password = 'ACCESS_TOKEN_GOES_HERE'
        }
    }
}

dependencies {
    implementation 'com.google.inject:guice:5.1.0'
    implementation "disorg.soliloquy:soliloquy.specs:0.0.397"
    implementation "disorg.soliloquy:inaugural.soliloquy.tools:0.0.93"
    implementation "disorg.soliloquy:inaugural.soliloquy.common:0.0.16"
    testImplementation "org.junit.jupiter:junit-jupiter-engine:5.9.0"
    testImplementation "org.junit.jupiter:junit-jupiter-api:5.9.0"
    testImplementation "org.junit.jupiter:junit-jupiter-params:5.9.0"
    implementation 'org.junit.jupiter:junit-jupiter-api:5.9.0'
    testImplementation 'junit:junit:4.13.2'
    implementation "org.mockito:mockito-core:4.8.0"
}

test {
    useJUnitPlatform()
}

publishing {
    repositories {
        maven {
            name = "inaugural.soliloquy.ruleset"
            url = uri("https://maven.pkg.github.com/felix-t-morgenstern/inaugural.soliloquy.ruleset")
            credentials {
                username = project.findProperty("gpr.user") ?: System.getenv("SOLILOQUY_GITHUB_PUBLISH_USERNAME")
                password = project.findProperty("gpr.user") ?: System.getenv("SOLILOQUY_GITHUB_PUBLISH_TOKEN")
            }
        }
    }
    publications {
        gpr(MavenPublication) {
            from(components.java)
        }
    }
}`

Have I set anything up incorrectly? I don't think I'm doing anything terribly exotic with Gradle here. It's a bit annoying to have to create a new PAT for my Gradle project after every time I push package updates in Maven. And, I know I should only be using one build tool across my projects; I was hoping I could transition from Maven to Gradle piecemeal, but if there's some reason why I should prioritize that, that's fine; I just want to know what I'm missing here.

Thanks!

**Comment by ossanna16** [comment]
Hi there @felix-t-morgenstern and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)
